### PR TITLE
Code Quality: Remove thumbnail cache to simplify logic and improve performance

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [ ] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #issue...

**Validation**
How did you test these changes?
- [ ] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 1]
```diff
@@ -937,117 +937,142 @@ private async Task<BitmapImage> GetShieldIcon()
...
+			var icon = await FileThumbnailHelper.GetIconAsync(
+					item.ItemPath,
+					thumbnailSize,
+					item.IsFolder,
+					true,
+					IconOptions.None);
+
+			// Loop until cached thumbnail is loaded or timeout is reached
+			var cancellationTokenSource = new CancellationTokenSource(3000);
+			while (!icon.isIconCached)
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1052]:
```suggestion
			while (true)
```

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 2]
```diff
@@ -937,117 +937,142 @@ private async Task<BitmapImage> GetShieldIcon()
...
+					IconOptions.None);
 
-				if (iconInfo.IconData is not null)
-				{
-					await dispatcherQueue.EnqueueOrInvokeAsync(async () =>
-					{
-						item.FileImage = await iconInfo.IconData.ToBitmapAsync();
-					}, Microsoft.UI.Dispatching.DispatcherQueuePriority.Low);
-				}
+				if (cancellationTokenSource.Token.IsCancellationRequested)
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1061]:
```suggestion
				if (icon.isIconCached || cancellationTokenSource.Token.IsCancellationRequested)
```
This prevents an extra 500 msec wait.

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 3]
```diff
@@ -1198,6 +1223,14 @@ public async Task LoadExtendedItemPropertiesAsync(ListedItem item)
...
 							});
 						}
+						else
+						{
+							// Load cached thumbnail for cloud files
+							if (item.SyncStatusUI.SyncStatus != CloudDriveSyncStatus.NotSynced && item.SyncStatusUI.SyncStatus != CloudDriveSyncStatus.Unknown)
+								_ = LoadCachedThumbnailAsync(item);
+							else
+								_ = LoadThumbnailAsync(item);
+						}
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1233]: Should this not be executed if `wasSyncStatusLoaded` is false? More to the point, I think this doesn't need to be executed after the properties are loaded, but could be executed immediately after the basic icon has been loaded.

Comment (yaira2) [reply src/Files.App/Data/Models/ItemViewModel.cs:1233]: > but could be executed immediately after the basic icon has been loaded.

We can execute `LoadThumbnailAsync` at that point, but then we'd also be executing `LoadCachedThumbnailAsync` for cloud files once sync status is loaded.

Comment (hishitetsu) [reply src/Files.App/Data/Models/ItemViewModel.cs:1233]: I see.

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 4]
```diff
@@ -1092,7 +1117,7 @@ public async Task LoadExtendedItemPropertiesAsync(ListedItem item)
 						}
 
 						cts.Token.ThrowIfCancellationRequested();
-						_ = LoadItemThumbnailAsync(item);
+						_ = LoadBasicIconAsync(item);
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1120]:
```suggestion
						await LoadBasicIconAsync(item);
```
It is better to await if the process is to be completed quickly,

File: src/Files.App/Data/Models/ItemViewModel.cs (hunk 3 above)
Comment (yaira2) [reply src/Files.App/Data/Models/ItemViewModel.cs:1233]: If you think that's fine, then we can go ahead and do make this change.

**Review by hishitetsu — APPROVED** [review]
LGTM in terms of code quality

File: src/Files.App/Data/Models/ItemViewModel.cs (hunk 3 above)
Comment (yaira2) [reply src/Files.App/Data/Models/ItemViewModel.cs]: I made additional changes that I think resolve this thread

File: src/Files.App/Data/Models/ItemViewModel.cs [hunk 5]
```diff
@@ -1198,6 +1098,15 @@ public async Task LoadExtendedItemPropertiesAsync(ListedItem item)
 								SetFileTag(item);
 							});
 						}
+						else
+						{
+							// Try loading thumbnail for cloud files in case they weren't cached the first time
+							if (item.SyncStatusUI.SyncStatus != CloudDriveSyncStatus.NotSynced && item.SyncStatusUI.SyncStatus != CloudDriveSyncStatus.Unknown)
+							{
+								await Task.Delay(500);
+								_ = LoadThumbnailAsync(item);
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/ItemViewModel.cs:1107]:
```suggestion
								_ = Task.Run(async () => {
									await Task.Delay(500);
									await LoadThumbnailAsync(item);
								});
```

**Pull request merged**